### PR TITLE
Bugfix/266 available space

### PR DESCRIPTION
**Pull request by Mortadha-abderrhim**
This PR deals with the case where there is not enough space in local storage.
In that case, an IOException is fired, and it is handled in the following manner:
1/ All files related to that experience are deleted.
2/ A toast is shown to state that the download failed.
3/ The experience is not added to the downloaded room.

! I am not sure how to test this case. Let me know if you have any suggestions.

### CODE REVIEW

File: app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt [hunk 1]
```diff
@@ -28,7 +28,11 @@ class RoomDownloadRepository @Inject constructor(
 
     override suspend fun addExperienceToDownloads(experience: ExperienceDto, context: Context) {
         val exp = downloaderService.downloadFiles(experience, context)
-        downloadDao.insertDownloaded(experienceToDownload(exp))
+        if (exp.type?.equals("failed") != true) downloadDao.insertDownloaded(
+            experienceToDownload(
```
Comment (amomo25) [inline app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt:32]: Pour la condition du if, je pense que ce serait mieux de faire not(exp.type?.equals("failed")) plutôt que de comparer avec true.

**Review by amomo25 — COMMENTED** [review]
The code looks great, I only have a small issue with the condition in the if branch.

File: app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt (hunk 1 above)
Comment (ToufiPF) [reply app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt:32]: actually you should be able to do exp.type != "failed" in kotlin

Comment (Mortadha-abderrhim) [reply app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt:32]: The problem is that it is a String? so it couldn't be done this way I think  `exp.type != "failed"`
I'll do it like @amomo25 suggested.

Comment (amomo25) [reply app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt:32]: No, in Kotlin you can compare Strings with == : https://www.tutorialkart.com/kotlin/compare-strings-in-kotlin-example/

Comment (Mortadha-abderrhim) [reply app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt:32]: The problem is in the ? (nullable) I think, but yes I'll try if it works.

Comment (Mortadha-abderrhim) [reply app/src/main/java/com/pindex/main/repositories/download/RoomDownloadRepository.kt]: That worked!

**Review by amomo25 — APPROVED** [review]

**Review by hadiyahya11 — DISMISSED** [review]
Nice PR. Concise and direct to the point.

**Review by amomo25 — DISMISSED** [review]

**Review by ToufiPF — DISMISSED** [review]

**Review by ToufiPF — APPROVED** [review]

**Review by amomo25 — APPROVED** [review]

**Pull request merged**